Quantized FC operator

jianyuh opened the pull request:
Summary:
Implements a basic quantized FC (uint8 * int8 -> uint8) with FBGEMM APIs.

Related document: https://fb.quip.com/rP5tAx56ApMM.

Work Item List:
1. [DONE] currently we use prepack routines inside Quantized FC operator. Will separate it as a standalone operator soon.
2. [Reverted] rebase to D14742020 (cpp custom types).
3. [DONE] correctness unit test.

4. [To Do] rebase to QTensor. Similar to D14565413, this will be implemented in the next Diff.

Differential Revision: D14761865

raghuramank100 commented on aten/src/ATen/native/quantized/cpu/qfc_prepack.cpp:
```diff
@@ -0,0 +1,108 @@
+
+    // ptr->printPackedMatrix("packedB inside qfc_prepack: ");
+
+    // We store this instance away in a Tensor and register a deleter function
```
Please add a TODO stating that we will need to replace this with torchscript classes at a later point.

jianyuh committed:
Quantized FC operator

Summary:
Implements a basic quantized FC (uint8 * int8 -> uint8) with FBGEMM APIs.

Related document: https://fb.quip.com/rP5tAx56ApMM.

Work Item List:
1. [DONE] currently we use prepack routines inside Quantized FC operator. Will separate it as a standalone operator soon.
2. [DONE] rebase to D14817809 and D14994781 (cpp custom types).
3. [DONE] correctness unit test.

4. [To Do] rebase to QTensor. Similar to D14565413, this will be implemented in the next Diff.

Differential Revision: D14761865

fbshipit-source-id: 5042f60169a2a4c37b7ca1ea1759b5058a583c3f
